### PR TITLE
Add logging to NGINX API http requests

### PR DESCRIPTION
**Pull request by dhurley**
### Proposed changes

- Add logging to NGINX API http requests
- Increase http client timeout

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [ ] I have read the [`CONTRIBUTING`](https://github.com/nginx/agent/blob/main/docs/CONTRIBUTING.md) document
- [ ] I have run ```make install-tools``` and have attached any dependency changes to this pull request
- [ ] If applicable, I have added tests that prove my fix is effective or that my feature works
- [ ] If applicable, I have checked that any relevant tests pass after adding my changes
- [ ] If applicable, I have updated any relevant documentation ([`README.md`](https://github.com/nginx/agent/blob/main/README.md))
- [ ] If applicable, I have tested my cross-platform changes on Ubuntu 22, Redhat 8, SUSE 15 and FreeBSD 13

### CODE REVIEW

File: sdk/config_helpers.go [hunk 1]
```diff
@@ -856,18 +856,21 @@ func nginxPlusApiCallback(parent *crossplane.Directive, current *crossplane.Dire
 }
 
 func pingStubStatusApiEndpoint(statusAPI string) bool {
-	client := http.Client{Timeout: 50 * time.Millisecond}
+	client := http.Client{Timeout: 1 * time.Second}
```
Comment (oliveromahony) [inline sdk/config_helpers.go:859]: can we make this configurable? Saves us from having to PR it again in the future

File: sdk/config_helpers.go [hunk 2]
```diff
@@ -882,27 +885,34 @@ func pingStubStatusApiEndpoint(statusAPI string) bool {
...
+		log.Debugf("Unable to read NGINX Plus API response body: %v", err)
 		return false
 	}
 
 	// Expecting API to return the api versions in an array of positive integers
 	// subset example: [ ... 6,7,8,9 ...]
 	var responseBody []int
 	err = json.Unmarshal(bodyBytes, &responseBody)
+	if err != nil {
+		log.Debugf("Unable to unmarshal NGINX Plus API response body: %v", err)
```
Comment (oliveromahony) [inline sdk/config_helpers.go:911]: Is this a warning?

File: sdk/config_helpers.go [hunk 3]
```diff
@@ -882,27 +885,34 @@ func pingStubStatusApiEndpoint(statusAPI string) bool {
...
 	}
 
 	if resp.StatusCode != 200 {
+		log.Debugf("NGINX Plus API responded with a %d status code", resp.StatusCode)
 		return false
 	}
 
 	bodyBytes, err := io.ReadAll(resp.Body)
 	if err != nil {
+		log.Debugf("Unable to read NGINX Plus API response body: %v", err)
```
Comment (oliveromahony) [inline sdk/config_helpers.go:902]: Is this a warning?

File: sdk/config_helpers.go [hunk 4]
```diff
@@ -856,18 +856,21 @@ func nginxPlusApiCallback(parent *crossplane.Directive, current *crossplane.Dire
...
 	}
 
 	if resp.StatusCode != 200 {
+		log.Debugf("Stub Status API responded with a %d status code", resp.StatusCode)
 		return false
 	}
 
 	bodyBytes, err := io.ReadAll(resp.Body)
 	if err != nil {
+		log.Debugf("Unable to read Stub Status API response body: %v", err)
```
Comment (oliveromahony) [inline sdk/config_helpers.go:873]: Is this a warning?

File: sdk/config_helpers.go [hunk 5]
```diff
@@ -856,18 +856,21 @@ func nginxPlusApiCallback(parent *crossplane.Directive, current *crossplane.Dire
 }
 
 func pingStubStatusApiEndpoint(statusAPI string) bool {
-	client := http.Client{Timeout: 50 * time.Millisecond}
+	client := http.Client{Timeout: 1 * time.Second}
 	resp, err := client.Get(statusAPI)
 	if err != nil {
+		log.Debugf("Unable to create client for Stub Status API request: %v", err)
```
Comment (oliveromahony) [inline sdk/config_helpers.go:862]: Is this a warning?

File: sdk/config_helpers.go [hunk 6]
```diff
@@ -882,27 +885,34 @@ func pingStubStatusApiEndpoint(statusAPI string) bool {
 }
 
 func pingNginxPlusApiEndpoint(statusAPI string) bool {
-	client := http.Client{Timeout: 50 * time.Millisecond}
+	client := http.Client{Timeout: 1 * time.Second}
 	resp, err := client.Get(statusAPI)
 	if err != nil {
+		log.Debugf("Unable to create client for NGINX Plus API request: %v", err)
```
Comment (oliveromahony) [inline sdk/config_helpers.go:891]: I think this is at least info level

**Review by olli-holmala — APPROVED** [review]

**Review by oliveromahony — APPROVED** [review]

**Review by aphralG — APPROVED** [review]

**Pull request merged**